### PR TITLE
drivers: i2c: cleanup CMakeLists.txt

### PR DESCRIPTION
**Pull request by ycsin**
Was poking around this area and feel like the `i2c/CMakeLists.txt` & `i2c/Kconfig` can probably use some housekeeping:
- Group similar things together and reorder alphabetically.
- Refactor RTIO driver Kconfig

If `I2C_RTIO` isn't enabled, there's only 1 driver implementation:

I2C driver Kconfig interface
Atmel SAM driver Kconfig interface

When `I2C_RTIO` is enabled, user can choose between 2 implementations:

I2C driver Kconfig interface
Atmel SAM driver Kconfig interface choice

### CODE REVIEW

File: drivers/i2c/Kconfig [hunk 1]
```diff
@@ -93,41 +93,41 @@ endif # I2C_RTIO
 
 # Include these first so that any properties (e.g. defaults) below can be
 # overridden (by defining symbols in multiple locations)
+source "drivers/i2c/Kconfig.ambiq"
```
Comment (MaureenHelm) [inline drivers/i2c/Kconfig:96]: Might as well add `zephyr-keep-sorted-start` and `zephyr-keep-sorted-stop` to keep them sorted

**Review by MaureenHelm — COMMENTED** [review]
@teburd please take a look

File: drivers/i2c/CMakeLists.txt [hunk 2]
```diff
@@ -4,80 +4,63 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/i2c.h)
...
+zephyr_library_sources_ifdef(CONFIG_I2C_GD32									i2c_gd32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_GECKO									i2c_gecko.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_INFINEON_CAT1					i2c_ifx_cat1.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_INFINEON_XMC4					i2c_ifx_xmc4.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_IMX										i2c_imx.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ITE_ENHANCE						i2c_ite_enhance.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ITE_IT8XXX2						i2c_ite_it8xxx2.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_LITEX									i2c_litex.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32									i2c_ll_stm32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1							i2c_ll_stm32_v1.c)
```
Comment (teburd) [inline drivers/i2c/CMakeLists.txt:37]: i2c_ll_stm32 is the top level driver file with implementation details in the v1 and v2, the older sources_ifdef made a bit more sense here

File: drivers/i2c/CMakeLists.txt [hunk 3]
```diff
@@ -4,80 +4,63 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/i2c.h)
...
+zephyr_library_sources_ifdef(CONFIG_I2C_MCUX									i2c_mcux.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NIOS2									i2c_nios2.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NPCX									i2c_npcx_controller.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NPCX									i2c_npcx_port.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NRFX_TWI							i2c_nrfx_twi.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NRFX_TWIM							i2c_nrfx_twim.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_NUMAKER								i2c_numaker.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_RCAR									i2c_rcar.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_RV32M1_LPI2C					i2c_rv32m1_lpi2c.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWI								i2c_sam_twi.c)
```
Comment (teburd) [inline drivers/i2c/CMakeLists.txt:55]: SAM_TWIHS is missing, as is the RTIO variant of the driver here it seems?

**Review by teburd — COMMENTED** [review]
I mean this cmake file could use some cleanup I don't disagree, but it seems in glancing at this that its missing some files build/options

File: drivers/i2c/CMakeLists.txt (hunk 2 above)
Comment (ycsin) [reply drivers/i2c/CMakeLists.txt:37]: The top level driver Kconfig [CONFIG_I2C_STM32](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_I2C_STM32) will be enabled when either `CONFIG_I2C_STM32_V1` / `CONFIG_I2C_STM32_V2` is enabled, the driver file is compiled in the line above (L36)

File: drivers/i2c/Kconfig (hunk 1 above)
Comment (ycsin) [reply drivers/i2c/Kconfig:96]: Will do for this file (if PR is acceptable) after a rebase, the CMakeLists.txt is a bit tricky as the Kconfig (`CONFIG_..` precedes filenames), I think sorting based on the file names looks cleaner.

File: drivers/i2c/CMakeLists.txt (hunk 3 above)
Comment (ycsin) [reply drivers/i2c/CMakeLists.txt:55]: I have created `CONFIG_I2C_SAM_TWI_RTIO` for the RTIO variant, it is currently in L64:
```
zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS_RTIO i2c_sam_twihs_rtio.c)
```

(The sorting of this file isn't quite right and needs a rebase, if we agree that this change is good to have)

File: drivers/i2c/CMakeLists.txt [hunk 4]
```diff
@@ -4,80 +4,63 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/i2c.h)
...
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM0									i2c_sam0.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIM							i2c_sam4l_twim.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SBCON									i2c_sbcon.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SC18IM704							i2c_sc18im704.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SEDI									i2c_sedi.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SIFIVE								i2c_sifive.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SMARTBOND							i2c_smartbond.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_TCA954X								i2c_tca954x.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS_RTIO				i2c_sam_twihs_rtio.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS							i2c_sam_twihs.c)
```
Comment (ycsin) [inline drivers/i2c/CMakeLists.txt:65]: @teburd 
(yeah, I know, the sorting order isn't quite right)

**Pull request closed without merging**